Test shutdown/reconfiguration mixture

eduard-bagdasaryan opened the pull request:
Added /signals command for sending a list of signals.
Protocol version 13.

eduard-bagdasaryan committed:
Test shutdown/reconfiguration mixture

Added /signals command (protocol version 13) to sends a
multiple signals to Squid at once (HUP and INT).

eduard-bagdasaryan commented on overlord.pl:
```diff
@@ -373,6 +377,40 @@ sub reconfigureSquid
+
+    my $signalNumber = @signalList;
+    &waitFor("Squid processed $signalListRaw signals", sub {
+        sleep($signalNumber*1); # estimated signal processing time: 1 second per signal
```
I could not find a more deterministic way for this waiting because we do not know beforehand whether Squid is going to process all N signals (queuing them), some of them or only the first one (ignoring N-1). We could make the estimation more precise, by running the existing reconfigureSquid() to work out the average time for reconfiguration (and use this time instead of 1 second), but it looks like an overkill.

rousskov replied:
I agree that running `reconfigureSquid()` to estimate reconfiguration delay is not a good solution.

Let's see whether we can get enough information by augmenting Squid to count and then, at `AvoidSignalAction()` time, show (at debug level 2) the number of received same-kind signals.

eduard-bagdasaryan committed:
Do not rely on delays while waiting for signals to be processed

Now Overlord expects Squid to log the number of received(processed)
signals:

    shutdown signals:N
    reconfiguration signals:N

and waits until the total logged number is equal to the expected
signals number.

eduard-bagdasaryan committed:
Send signals with a small delay to ensure their delivery

Matches Squid 0795071.
